pkg/controller/handlers/toolreference/toolreference.go

```diff
@@ -253,3 +296,16 @@ func (h *Handler) Populate(req router.Request, resp router.Response) error {
 
 	return nil
 }
+
+func (h *Handler) RemoveCredential(req router.Request, _ router.Response) error {
```

Comment:
nit: I think this should be renamed for greater clarity.

```suggestion
func (h *Handler) RemoveModelProviderCredential(req router.Request, _ router.Response) error {
```

Reply:
Done.